@xen-orchestra/audit-core/src/index.spec.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-env jest */
+
+import { AuditCore, NULL_ID, Storage } from '.'
+
+const asyncIteratorToArray = async asyncIterator => {
+  const array = []
+  for await (const entry of asyncIterator) {
+    array.push(entry)
+  }
+  return array
+}
+
+class DB extends Storage {
+  constructor() {
+    super()
+
+    this._db = new Map()
+    this._lastId = undefined
+  }
+
+  async put(record) {
+    this._db.set(record.id, record)
+  }
+
+  async setLastId(id) {
+    this._lastId = id
+  }
+
+  async getLastId() {
+    return this._lastId
+  }
+
+  async del(id) {
+    this._db.delete(id)
+  }
+
+  async get(id) {
+    return this._db.get(id)
+  }
+}
+
+const DATA = [
+  [
+    {
+      id: '87b98aae-32c6-11ea-978f-2e728ce88125',
+      ip: '192.168.100.212',
+      name: 'toto',
+    },
+    'add',
+    {
+      parameters: {
+        name: 'resource',
+        id: 'b40f49ea-32c6-11ea-aec2-2e728ce88125',
+      },
+      result: 'success',
+    },
+  ],
+  [
+    {
+      id: '87b98d2e-32c6-11ea-978f-2e728ce88125',
+      ip: '192.168.100.5',
+      name: 'titi',
+    },
+    'del',
+    { result: 'success' },
+  ],
+  [
+    {
+      id: '87b98e8c-32c6-11ea-978f-2e728ce88125',
+      ip: '192.168.100.111',
+      name: 'foo',
+    },
+    'del',
+    {
+      result: 'success',
+    },
+  ],
+]
```

Comment:
The data set seems a bit too complex for debugging when tests fail.
What about:
```js
[
  {
    { name: 'subject0' },
    'event0',
    {}
  },
  {
    { name: 'subject1' },
    'event1',
    {}
  },
  {
    { name: 'subject2' },
    'event2',
    {}
  },
]
```